dmoj/checkers/bridged.py

```diff
@@ -0,0 +1,122 @@
+import os
+import traceback
+import tempfile
+
+from dmoj.error import CompileError, InternalError
+from dmoj.executors import executors
+from dmoj.executors.base_executor import CompiledExecutor
+from dmoj.judgeenv import env, get_problem_root
+from dmoj.result import CheckerResult
+from dmoj.utils import ansi
+from dmoj.utils.unicode import utf8text
+
+checker_defaults = {
+    'time_limit': env['generator_time_limit'],
+    'memory_limit': env['generator_memory_limit'],
+    'compiler_time_limit': env['compiler_time_limit'],
+    'feedback': True
+}
+
+executor = None
+
+def get_executor(checker_kwargs, problem_id):
+    global executor
+
+    if executor is None:
+        if 'files' not in checker_kwargs:
+            raise InternalError('No checker file[s] specified!')
+        if 'lang' not in checker_kwargs:
+            raise InternalError('Language not specified for checker!')
+
+        filenames = list(map(lambda x: os.path.join(get_problem_root(problem_id), x), checker_kwargs['files']))
```

Comment:
Given this shares a lot of logic with the generator, should we move this into a `dmoj.utils` file?

Reply:
yes